src/Catalyst.TestUtils/DiscoveryHelper.cs

```diff
@@ -0,0 +1,244 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Concurrency;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.IO.Messaging.Correlation;
+using Catalyst.Abstractions.Network;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Abstractions.P2P.Discovery;
+using Catalyst.Abstractions.P2P.IO.Messaging.Correlation;
+using Catalyst.Abstractions.P2P.IO.Messaging.Dto;
+using Catalyst.Abstractions.Types;
+using Catalyst.Abstractions.Util;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.IO.Messaging.Correlation;
+using Catalyst.Core.Lib.Network;
+using Catalyst.Core.Lib.P2P.Discovery;
+using Catalyst.Core.Lib.P2P.IO.Messaging.Correlation;
+using Catalyst.Core.Lib.P2P.Models;
+using Catalyst.Core.Lib.P2P.Repository;
+using Catalyst.Core.Lib.P2P.ReputationSystem;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Core.Modules.P2P.Discovery.Hastings;
+using Catalyst.Protocol.Peer;
+using DnsClient;
+using Microsoft.Extensions.Caching.Memory;
+using Nethereum.Hex.HexConvertors.Extensions;
+using NSubstitute;
+using Serilog;
+using SharpRepository.InMemoryRepository;
+
+namespace Catalyst.TestUtils
+{
+    public static class DiscoveryHelper
+    {
+        public sealed class DevDiscover : BaseDiscovery
+        {
+            public override Task DiscoveryAsync()
+            {
+                return Task.Run(async () =>
+                {
+                    while (true)
+                    {
+                        // don't run again for at least 200 milliseconds
+                        await Task.Delay(200);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)